lib/bundler/definition.rb

```diff
@@ -137,8 +137,17 @@ def resolve_remotely!
     # @return [Bundler::SpecSet]
     def specs
       @specs ||= begin
-        specs = resolve.materialize(Bundler.settings[:cache_all_platforms] ? dependencies : requested_dependencies)
-
+        begin
+          specs = resolve.materialize(Bundler.settings[:cache_all_platforms] ? dependencies : requested_dependencies)
+        rescue GemNotFound => e # Handle yanked gem
+          gem_info = extract_gem_info(e)
+          locked_gem = @locked_specs[gem_info[0]].last
```

Comment:
I'd like to see some more explicitness about what information about the gem you're extracting (maybe through variable assignment?). What is being pulled out of `gem_info`?